geth/proxy/transactions/sendtransaction.go

```diff
@@ -36,15 +38,32 @@ func ExecuteRemoteSendTransaction(manager common.RPCNodeManager, req common.RPCC
 		return nil, err
 	}
 
-	nonce := uint64(req.ID)
-	toAddr := req.ParseFromAddress()
+	client, err := manager.RPCClient()
+	if err != nil {
+		return nil, err
+	}
+
+	fromAddr := req.ParseFromAddress()
+
+	// We need to request a new transaction nounce from upstream node.
+	ctx, canceler := context.WithDeadline(context.Background(), time.Now().Add(1*time.Minute))
+
+	defer canceler()
```

Comment:
Yes, though in this case the call to `CallContext` will block execution till response is received, but still ensure we cancel the context at the end of the function to avoid go-routine leakage is necessary, else we use `Call`, which am not sure uses a timeout. Will have to confirm.